File: qb-engineer-ui/src/app/features/invoices/models/create-invoice-request.model.ts
import { CreateInvoiceLineRequest } from './create-invoice-line-request.model';

export interface CreateInvoiceRequest {
  customerId: number;
  salesOrderId?: number;
  shipmentId?: number;
  invoiceDate: string;
  dueDate: string;
  creditTerms?: string;
  taxRate: number;
  notes?: string;
  lines: CreateInvoiceLineRequest[];
}

File: qb-engineer-server/qb-engineer.core/Models/CreateInvoiceRequestModel.cs
namespace QBEngineer.Core.Models;

public record CreateInvoiceRequestModel(
    int CustomerId,
    int? SalesOrderId,
    int? ShipmentId,
    DateTimeOffset InvoiceDate,
    DateTimeOffset DueDate,
    string? CreditTerms,
    decimal TaxRate,
    string? Notes,
    List<CreateInvoiceLineModel> Lines);

public record CreateInvoiceLineModel(
    int? PartId,
    string Description,
    int Quantity,
    decimal UnitPrice);
